CloudFormation strategy

aarongorka opened the pull request.

Commit:
Add function to return next free priority

Commit:
Add functionality to deploy CloudFormation template

Remove unnecessary functions

Commit:
Create task definition separately to CloudFormation

Commit:
Remove tabs, fix closing parenthesis

adenot commented on scripts/deploy.py:
```diff
@@ -173,103 +139,86 @@ def get_alb_target_group(cluster, name):
+    parameters = {}  # what parameters will the template have?
 
-        wait_ecs_service(name_public)
+    stack_name = "MV-{env}-{app_name}-{version}-{realm}".format(env=os.environ['ENV'], app_name=config['ecs_app_name'], version=os.environ['BUILD_VERSION'], realm=os.environ['REALM'])
```
should it be `MV-{realm}-{app_name}-{version}-{env}`

Commit:
Add environment var list generator, remove jinja, fix typos

Commit:
Remove all old functions, fix typos, add cutover script

Commit:
Add service template

Commit:
Remove ServiceName , PlacementConstraint and DeploymentConfiguration

Defining ServiceName doesn't allow for updates on the CloudFormation
template. PlacementConstraints is unnecessary and default values for
DeploymentConfiguration are better.

adenot committed:
Adding Deregistration delay timeout parameter in the TargetGroup

adenot committed:
Autoscaling for ECS Service

Commit:
Fix typo

Commit:
Working deployment script

Commit:
Add example app

adenot committed:
DNS name for internal applications

adenot committed:
Merge branch 'feature/cloudformation-strategy' of github.com:amaysim-au/docker-ecs-utils into feature/cloudformation-strategy

Commit:
Don't raise error

Commit:
Merge branch 'feature/cloudformation-strategy' of github.com:amaysim-au/docker-ecs-utils into feature/cloudformation-strategy

Commit:
Working Docker deployment

Updated Dockerfile:
  * Reduced number of layers by combining `apk` commands
  * No need to clean up with `--no-cache`
  * Removed Python (already installed from base image)
  * Removed `build_deps` and `libintl` (?)
  * Removed other unused packages (less, curl, etc.)
  * Combined all  `RUN` directives in to a single command
  * Change WORKDIR to `/srv/app` so we're not mounting to /
  * Removed `requriements.txt` since we're only installing two packages

Added example deployment

Small refactor on Python scripts, reordered some parts for better
failure handling

Removed ruamel.yaml as a dependency. Difficult to install in Docker and
the default YAML parser works fine.

Commit:
Update .env.template in example

Commit:
Update .gitignore with interpolated ecs-config.yml
